### ISSUE TITLE
User input filter does not open on iphone 

### ISSUE THREAD

**Issue by skrishn** [Esri/solutions-components#1004]
The user input filter does not show the drop-down on an iphone. Works on small screens on DT and android.

to repro:

Create a reporter app
Add a user input filter to a layer
open app on an iphone
go to layer with filter and try to open

![image](https://github.com/user-attachments/assets/90522158-355f-463b-bd23-f83e73d3758e)

sample app: https://nw-brews.mapsdevext.arcgis.com/apps/instant/reporter/index.html?appid=2847af40c46e4d2cbd65476e7ca30029
Go to weed observations and open the filter on an iphone.

For mobile only, we'd like to go back to takeover modal. 

![image](https://github.com/user-attachments/assets/f5890139-af43-4a8f-a5f2-5c645124bc96)
Production app https://localdeployment.maps.arcgis.com/apps/instant/reporter/index.html?appid=3f6358973ca44301ad2d90f4919e7827

**Comment by skrishn** [comment]
@sumitzarkar We're moving it to R1.

**Comment by sumitzarkar** [comment]
@skrishn 

- This is the instant-app-filterlist component, and the dropdown control is rendered and controlled by the component
- This looks like a default behaviour of the control in iPhone
- We tested with nearby instant app and found same behaviour
- This because the keypad opens in iPhone on clicking the control if you hide the keypad you will see the dropdown

https://github.com/user-attachments/assets/d469c8a3-6a2b-495c-9988-cd416ed6beb3

**Comment by sumitzarkar** [comment]
@skrishn 

Please let us know any any changes requried from our end on this

**Comment by skrishn** [comment]
@sumitzarkar Yes, we;d require changes and I will update on this end of this week after doing some tests.

**Comment by skrishn** [comment]
@sumitzarkar Could you share the app used in the video above? https://github.com/Esri/solutions-components/issues/1004#issuecomment-2458714439

**Comment by sumitzarkar** [comment]
@skrishn 
> Could you share the app used in the video above? https://github.com/Esri/solutions-components/issues/1004#issuecomment-2458714439

Shared the app link on Teams

**Comment by skrishn** [comment]
@sumitzarkar PE is still testing the issue, will update you

**Comment by chris-fox** [comment]
@skrishn, @sumitzarkar, we discussed this on the call with the Instant Apps team today, they have been unable to reproduce the issue with nearby. The recommendation was to move forward with moving the filter to a full screen modal on mobile devices only. This would be consistent with other apps and seems like a reasonable fix. The drawback of being in the modal is it doesn't allow you to view the applied filter dynamically in the map, but you can see the map in mobile mode anyway so there doesn't seem to be a drawback to this implementation.

**Comment by pratiksabale07** [comment]
@skrishn @chris-fox 

- The filter list should be shown in a **modal** for mobile view.
![image](https://github.com/user-attachments/assets/138c473b-3e35-4d3d-9239-44ca4d9c8e52)
 
- The filter list is displayed in a **side panel** for desktop view.
- So while switching from mobile to desktop/desktop to mobile mode when need to show the appropriate filter (to maintain responsive behaviour)
## Approach
### Switching from Mobile to Desktop Mode
1. We have a mobile watch handler that returns a false value when we switch to the desktop mode
2. Close the filter modal to save the current filter settings (layer expressions).  
3. Render the filter panel in the side panel
4. Pass the current filter settings to the `instant-app-filter-list` to retain the applied filters.
### Switching from Desktop to Mobile Mode
1. We have a mobile watch handler which returns true value when we switch to the mobile mode
2. Close the side panel to save the current filter settings (layer expressions).  
3. Open the filter modal and pass the current filter settings to the `instant-app-filter-list`.  
4. The filter list will now appear in the modal with applied filters retained.

**Comment by pratiksabale07** [comment]
@skrishn @chris-fox 
## Cause
- The filter input was not opening on the iPhone
- https://github.com/Esri/solutions-components/issues/1004#issuecomment-2458714439
___
## Solution
- Display filter list in a modal for mobile view.
- Display filter list in the side panel for desktop view
- https://github.com/Esri/solutions-components/issues/1004#issuecomment-2591676238
___
## Dev Checklist
Items  | Is required | Status | Comments
-- | -- | -- | --
Required NLS changes? | No   |   | 
Required Backward compatibility | No  |   | 
Required 508 | Yes  |   | 
Need Sanitize? | No  |   | 
___
## Impact Analysis Report
Impacted Areas | Comments
-- | --
Have Configuration changes? | No
Have Runtime changes? | Yes
Impact on RTL | Yes
Impact on runtime | - The filter list should be displayed in a modal for mobile view.
|| - The filter list should be displayed in a side panel for desktop view.
|| - While switching from mobile to desktop/desktop to mobile mode the appropriate filter (to maintain responsive behaviour) should be shown.
